plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java

```diff
@@ -286,4 +314,96 @@ public void dropSchema(ConnectorSession session, String schemaName)
     {
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support dropping schemas");
     }
+
+    private Set<AggregateFunctionRule> aggregateFunctionRules()
+    {
+        JdbcTypeHandle bigIntHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), 0, 0, Optional.empty(), Optional.empty());
+        JdbcTypeHandle doubleHandle = new JdbcTypeHandle(Types.DOUBLE, Optional.of("double"), 0, 0, Optional.empty(), Optional.empty());
+        Set<Type> numericTypes = Set.of(DOUBLE, BIGINT, REAL);
+        Set<Type> numericAndTimeStampType = Set.of(DOUBLE, BIGINT, REAL, TimestampType.createTimestampType(3));
+        Set<Type> allTypes = Set.of(DOUBLE, BIGINT, REAL, TimestampType.createTimestampType(3), VARCHAR);
+        ImmutableSet.Builder<AggregateFunctionRule> builder = ImmutableSet.<AggregateFunctionRule>builder()
+                .add(new ImplementCountAll(bigIntHandle))
+                .add(new ImplementCount(bigIntHandle));
+
+        allTypes.forEach(type ->
+                builder.add(SingleInputAggregateFunction.builder()
+                        .prestoName("approx_distinct")
+                        .expression("approx_count_distinct(%s)")
+                        .jdbcTypeHandle(bigIntHandle)
+                        .outputType(BIGINT)
```

Comment:
isn't output type be a function of the type handle?

or you mean this to be part of the match (function output)?
then, isn't -- logically -- type handle a function of the output type? (see `io.trino.plugin.postgresql.PostgreSqlClient#toTypeHandle`)

Reply:
outputType is part of the match , jdbcTypeHandle seem to be able to capture more than the type though so I kept both explicitly but I am not really using all the other properties of jdbcTypeHandle yet. If you prefer I use only one of those and derive the other let me know.